wrappers/IWear/src/IWearWrapper.cpp

```diff
@@ -138,13 +138,20 @@ void IWearWrapper::run()
         return;
     }
 
-    if (pImpl->iWear->getStatus() != WearStatus::Ok) {
+    if (pImpl->iWear->getStatus() == WearStatus::Error
+            || pImpl->iWear->getStatus() == WearStatus::Unknown) {
         yError() << logPrefix << "The status of the IWear interface is not Ok"
                  << static_cast<int>(pImpl->iWear->getStatus()) << ")";
         askToStop();
         return;
     }
 
+    // case status is TIMEOUT or DATA_OVERFLOW
+    if (pImpl->iWear->getStatus() != WearStatus::Ok) {
```

Comment:
> How are the cases like `CALIBRATING` and `DATA_OVERFLOW` handled in this case. They should stop the device and call `askToStop()`.

The `Calibrating` and `WaitingForFristRead` status are already handled previously in line https://github.com/lrapetti/wearables/blob/fix-IWearWrapperRemapper-status/wrappers/IWear/src/IWearWrapper.cpp#L132.



> Furthermore, in case of timeout, it is better to implement at the wearable device level some method to wait during the timeout for a duration that can be set through parameter. For e.g., in IAnalogSensorToIWear wearable device, the [getStatus()](https://github.com/robotology/wearables/blob/ba85e3f9dced710b0535ebe5506c8a23c3d4271d/devices/IAnalogSensorToIWear/src/IAnalogSensorToIWear.cpp#L584) method should handle the case of timeout by waiting for some time.
> 
> @diegoferigo what do you think ?

Regarding this, what discussed with @diegoferigo was that the `IWearWrapper` should stream the timeout status as it is without waiting, since its role is to remap the data with a common (wearable) format. The logic for checking/waiting if the status goes back to `OK` can then be implemented in the consumer device.

Reply:
> The Calibrating and WaitingForFristRead status are already handled previously in line https://github.com/lrapetti/wearables/blob/fix-IWearWrapperRemapper-status/wrappers/IWear/src/IWearWrapper.cpp#L132.

my bad! :sweat_smile:

Reply:
> IWearWrapper should stream the timeout status as it is without waiting, since its role is to remap the data with a common (wearable) format

I agree now

Reply:
> The logic for checking/waiting if the status goes back to OK can then be implemented in the consumer device.

:+1: